Src/Newtonsoft.Json.Tests/Linq/AnnotationsTests.cs

```diff
@@ -308,6 +308,21 @@ public void MultipleAnnotationsAreCopied()
             Assert.AreEqual(annotation, t.DeepClone().Annotation<T>());
         }
 
+        [Test]
+        public void MultipleAnnotationsAreNotCopiedWithSetting()
```

Comment:
the code changes look good.
As we are updating all the other derivates of JToken, we should add case for each one in the test. I think a single unit test for all should be sufficient, let's just clone each type in the test and verify.

Reply:
Added. Also, the test passes.